### PR TITLE
[wasm] Allow assemblies to be skipped for AOTMode=AotInterp

### PR DESCRIPTION
**Pull request by radical**
* Allow skipping AOT, per assembly with %(_InternalForceInterpret) metadata
* new `$(WasmNativeDebugSymbols)` to control symbols for `dotnet.wasm` (useful only for AOT case), which defaults to true
* Path handling fixes for AOT builds on windows
* add create-packs.proj for setting up packs from a local build, for quick local testing

* Also, note that this changes the default `$(AOTMode)` to `AotInterp`

### CODE REVIEW

**Comment by radical** [comment]
This is essentially https://github.com/dotnet/runtime/pull/48500 minus the renaming of public items/properties.

File: src/mono/wasm/build/WasmApp.targets [hunk 1]
```diff
@@ -264,39 +292,52 @@
...
 
-     <_WasmObjects Include="$(_WasmIntermediateOutputPath)driver.o"/>
-     <_WasmObjects Include="$(_WasmIntermediateOutputPath)pinvoke.o"/>
-     <_WasmObjects Include="$(_WasmIntermediateOutputPath)corebindings.o"/>
+     <_WasmObjectsToBuild Include="$(_WasmIntermediateOutputPath)driver.o"/>
+     <_WasmObjectsToBuild Include="$(_WasmIntermediateOutputPath)pinvoke.o"/>
+     <_WasmObjectsToBuild Include="$(_WasmIntermediateOutputPath)corebindings.o"/>
+     <_WasmObjectsToBuild SourcePath="$(_WasmSrcDir)%(FileName).c" />
+
+     <_WasmObjects Include="@(_WasmRuntimePackNativeLibs->'$(MicrosoftNetCoreAppRuntimePackRidNativeDir)%(FileName)%(Extension)')" />
```
Comment (lewing) [inline src/mono/wasm/build/WasmApp.targets:301]: Eventually we're going to want to pull these from the correctly versioned package directories too.

Comment (lewing) [reply src/mono/wasm/build/WasmApp.targets:301]: And now that I think about it, we'll probably need to add js from packages to _DotnetJSSrcFile as well.

File: src/mono/wasm/build/WasmApp.targets [hunk 2]
```diff
@@ -96,6 +102,10 @@
       <AOTMode Condition="'$(AOTMode)' == ''">LLVMOnly</AOTMode>
```
Comment (lewing) [inline src/mono/wasm/build/WasmApp.targets:102]:
```suggestion
      <AOTMode Condition="'$(AOTMode)' == ''">AotInterp</AOTMode>
```

File: src/mono/wasm/build/WasmApp.targets [hunk 3]
```diff
@@ -264,39 +292,52 @@
...
-     Text="Bug: Number of assemblies doesn't match the number of generated bitcode files. BitcodeFiles: @(_BitcodeFile->Count()) vs Assemblies: @(_WasmAssemblies->Count())" />
+   <Error Condition="'$(RunAOTCompilation)' == 'true' and @(_BitcodeFile->Count()) != @(_AOTAssemblies->Count())"
+          Text="Bug: Number of aot assemblies doesn't match the number of generated bitcode files. BitcodeFiles: @(_BitcodeFile->Count()) vs Assemblies: @(_AOTAssemblies->Count())" />
 
    <PropertyGroup>
-     <_WasmIncludeDir>$(MicrosoftNetCoreAppRuntimePackRidDir)native/include</_WasmIncludeDir>
-     <_WasmSrcDir>$(MicrosoftNetCoreAppRuntimePackRidDir)native/src</_WasmSrcDir>
-     <EmccCFlags>$(EmccFlags) -DCORE_BINDINGS -DGEN_PINVOKE=1 -I$(_WasmIntermediateOutputPath) -I$(_WasmIncludeDir)/mono-2.0 -I$(_WasmIncludeDir)/wasm</EmccCFlags>
+     <EmccCFlags>$(EmccFlags) -DCORE_BINDINGS -DGEN_PINVOKE=1 "-I$(_WasmIncludeDir)/mono-2.0" "-I$(_WasmIncludeDir)/wasm"</EmccCFlags>
+     <EmccCFlags Condition="'$(WasmNativeDebugSymbols)' == 'true'">$(EmccCFlags) -g</EmccCFlags>
```
Comment (lewing) [inline src/mono/wasm/build/WasmApp.targets:318]: We should add more control over the debug setup to MonoAOTCompiler in a follow up.

**Review by lewing — APPROVED** [review]
Thanks for separating this out.  Added some minor comments but I'm ok with it landing as is.

**Comment by steveisok** [comment]
This PR is important to fast track in order to help alleviate longer helix wasm runs.

**Pull request merged**